odoo/addons/base/tests/test_res_lang.py

```diff
@@ -40,3 +51,41 @@ def test_00_intersperse(self):
         assert intersperse("abc12", [3], '.') == ('abc12', 0)
         assert intersperse("abc12", [2], '.') == ('abc12', 0)
         assert intersperse("abc12", [1], '.') == ('abc1.2', 1)
+
+    def test_res_lang_decimal_point_preview(self):
+        self.assertEqual(self.lang_0.decimal_point_preview, '(e.g. "99.00")')
+        self.lang_0.write({'decimal_point': ','})
+        self.assertEqual(self.lang_0.decimal_point_preview, '(e.g. "99,00")')
+
+    def test_res_lang_thousand_sep_preview(self):
+        self.assertEqual(self.lang_0.thousand_sep_preview, '(e.g. "999,999,999")')
+        self.lang_0.write({'thousands_sep': '.'})
+        self.assertEqual(self.lang_0.thousand_sep_preview, '(e.g. "999.999.999")')
+
+    @freeze_time("2020-01-01 16:00:00")
+    def test_res_lang_date_format_preview(self):
+        lang_0, lang_1 = self.env['res.lang'].with_context(active_test=False).search(
+            [('iso_code', 'in', ['en', 'fr'])]
+        )
+
+        self.assertEqual(lang_0.date_format_preview, '(e.g. "01/01/2020")')
+        self.assertEqual(lang_1.date_format_preview, '(e.g. "01/01/2020")')
+
+        (lang_0 | lang_1).write({'date_format': '%A-%B-%Y'})
+
+        self.assertEqual(lang_0.date_format_preview, '(e.g. "Wednesday-January-2020")')
+        self.assertEqual(lang_1.date_format_preview, '(e.g. "mercredi-janvier-2020")')
+
+    @freeze_time("2020-01-01 16:00:00")
+    def test_res_lang_time_format_preview(self):
+        lang_0, lang_1 = self.env['res.lang'].with_context(active_test=False).search(
+            [('iso_code', 'in', ['en', 'fr'])]
+        )
+
+        self.assertEqual(lang_0.time_format_preview, '(e.g. "16:00:00")')
+        self.assertEqual(lang_1.time_format_preview, '(e.g. "16:00:00")')
+
+        (lang_0 | lang_1).write({'time_format': '%M-%I-%S'})
```

Comment:
(same above for other tests)

```suggestion
        (lang_0 | lang_1).time_format = '%M-%I-%S'